R/zchunk_LB1322.Fert.R

```diff
@@ -41,92 +43,396 @@ module_energy_LB1322.Fert_DISABLED <- function(command, ...) {
+    L1322.Fert_ALL_MtN_R_F_Y_adj %>%
+      select(GCAM_region_ID, sector, fuel, year, value = intensity_GJkgN) ->
+      L1322.IO_R_Fert_F_Yh # Note that this is a final output table
+
+    # Creating final output table "Adjusted industrial feedstock use by GCAM region / fuel / year"
+    # We already calcuated the values. We only need to add a column for the sector and strip out unnecessary columns.
+    L1322.Fert_ALL_MtN_R_F_Y_adj %>%
+      mutate(sector = "industry_feedstocks") %>%
+      select(GCAM_region_ID, sector, fuel, year, value = in_indfeed_netFert) ->
+      L1322.in_EJ_R_indfeed_F_Yh # Note that this is a final output table
+
+    # Extract the values we calculated for industrial energy use. These values will be used for its respective final table.
+    L1322.Fert_ALL_MtN_R_F_Y_adj %>%
+      select(GCAM_region_ID, fuel, year, value = in_indenergy_netFert) ->
+      L1322.in_EJ_R_indenergy_Ffert_Yh
+
+    # Creating final output table "Adjusted industrial energy use by GCAM region / fuel / year"
+    # We are starting with the input industrial energy table and replacing a portion of the values we calcuated in this chunk.
+    L1321.in_EJ_R_indenergy_F_Yh %>%
+      rename(value_indenergy = value) %>%
+      # left_join_error_no_match cannot be used because the joining tibble has less rows, so NAs will be introduced.
+           # This will be helpful as a filter in the following step.
+      left_join(L1322.in_EJ_R_indenergy_Ffert_Yh, by = c("GCAM_region_ID", "fuel", "year")) %>% # Join values we calculated
+      # Replace any NAs with original industrial energy value
+      mutate(value = replace(value, is.na(value), value_indenergy[is.na(value)])) %>%
+      select(GCAM_region_ID, sector, fuel, year, value) ->
+      L1322.in_EJ_R_indenergy_F_Yh # Note that this is a final output table.
+
+    # -----------------------------------------------------------------------------------------------------------------
+
+    # Calculate fertilizer non-energy costs by technology
+         # These technologies include gas, gas with CCS, coal, coal with CCS, and oil
+
+    # First, calculate gas cost per kg N
+         # Calculating non-energy costs for gas technology as USA market fertilizer price minus GCAM fuel costs
+         # Calculate the gas price as the sum of resource costs plus intermediate sectoral mark-ups
+         # 2005 is used as the fertilizer base price. Interpolate cost tables to make sure this year is included
+    A10.rsrc_info %>%
+      filter(resource == "natural gas") %>%
+      summarise(value = sum(`2005`)) %>% # Ensuring no duplicates
+      .[["value"]] -> # Save cost as single number. Units are 1975 USD per GJ.
+      A10.rsrc_info_2005
+
+    # Using primary energy handling default costs in 1971 and 2100, interpolate for 2005 for natural gas
+    A21.globaltech_cost %>%
+      mutate(`2005` = NA) %>% # Include placeholder for 2005. Value will be interpolated.
```

Comment:
Why is 2005 being interpolated? It would be good to define 2005 as a constant, probably, and then refer to it throughout next 40 lines of code.